runtime/runtime_controller.cc

```diff
@@ -115,11 +113,13 @@ std::unique_ptr<RuntimeController> RuntimeController::Clone() const {
 }
 
 bool RuntimeController::FlushRuntimeStateToIsolate() {
-  // TODO(dkwingsmt): Needs a view ID here (or platform_data should probably
-  // have multiple view metrics).
-  return SetViewportMetrics(kFlutterImplicitViewId,
-                            platform_data_.viewport_metrics) &&
-         SetLocales(platform_data_.locale_data) &&
+  for (auto const& [view_id, viewport_metrics] :
+       platform_data_.viewport_metrics_for_views) {
+    if (!AddView(view_id, viewport_metrics)) {
```

Comment:
Shouldn't this set the viewport metrics instead of adding the view? Wouldn't this cause errors if `RuntimeController::FlushRuntimeStateToIsolate` is called multiple times?

Reply:
Currently I don't see why `RuntimeController::FlushRuntimeStateToIsolate` will be called multiple times because it is only called when the root isolate becomes ready. I can add a boolean for it.

Reply:
Also this needs to add the view because for `PlatformDispatcher` no views have been added at all; it has just been constructed.

Reply:
Ahh interesting, I didn't realize `AddView` was ignored before the isolate is ready. Could we assert that `RuntimeController::FlushRuntimeStateToIsolate` is called no more than once?